lib/libexif/ExifParse.cpp

```diff
@@ -823,6 +823,15 @@ void CExifParse::GetLatLong(
     {
       Values[a] = ConvertAnyFormat(ValuePtr+a*ComponentSize, Format);
     }
+
+    // fix incorrect values to prevent buffer overflow (happening sometimes with Panasonic TZ40)
+    if (Values[0] > 180 || Values[1] > 60 || Values[2] > 60)
```

Comment:
I prefer just increase the string buffer length to overcome this. this validation line is indeed invalid.

on the other hand, we may consider value 16777215 as the invalid value, then give a non-fatal error and do nothing.